kustomize/internal/commands/edit/remove/removetransformer_test.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package remove
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+
+	"sigs.k8s.io/kustomize/api/filesys"
+	testutils_test "sigs.k8s.io/kustomize/kustomize/v3/internal/commands/testutils"
+)
+
+func TestRemoveTransformers(t *testing.T) {
+	type given struct {
+		transformers []string
+		removeArgs   []string
+	}
+	type expected struct {
+		transformers []string
+		deleted      []string
+		err          error
+	}
+	testCases := []struct {
+		description string
+		given       given
+		expected    expected
+	}{
+		{
+			description: "remove transformer",
+			given: given{
+				transformers: []string{
+					"transformer1.yaml",
+					"transformer2.yaml",
+					"transformer3.yaml",
+				},
+				removeArgs: []string{"transformer1.yaml"},
+			},
+			expected: expected{
+				transformers: []string{
+					"transformer2.yaml",
+					"transformer3.yaml",
+				},
+				deleted: []string{
+					"transformer1.yaml",
+				},
+			},
+		},
+		{
+			description: "remove transformers with pattern",
+			given: given{
+				transformers: []string{
+					"foo/transformer1.yaml",
+					"foo/transformer2.yaml",
+					"foo/transformer3.yaml",
+					"do/not/deleteme/please.yaml",
+				},
+				removeArgs: []string{"foo/transformer*.yaml"},
+			},
+			expected: expected{
+				transformers: []string{
+					"do/not/deleteme/please.yaml",
+				},
+				deleted: []string{
+					"foo/transformer1.yaml",
+					"foo/transformer2.yaml",
+					"foo/transformer3.yaml",
+				},
+			},
+		},
+		{
+			description: "nothing found to remove",
+			given: given{
+				transformers: []string{
+					"transformer1.yaml",
+					"transformer2.yaml",
+					"transformer3.yaml",
+				},
+				removeArgs: []string{"foo"},
+			},
+			expected: expected{
+				transformers: []string{
+					"transformer2.yaml",
+					"transformer3.yaml",
+					"transformer1.yaml",
+				},
+			},
+		},
+		{
+			description: "no arguments",
+			given:       given{},
+			expected: expected{
+				err: errors.New("must specify a transformer file"),
+			},
+		},
+		{
+			description: "remove with multiple pattern arguments",
+			given: given{
+				transformers: []string{
+					"foo/foo.yaml",
+					"bar/bar.yaml",
+					"transformer3.yaml",
+					"do/not/deleteme/please.yaml",
+				},
+				removeArgs: []string{
+					"foo/*.*",
+					"bar/*.*",
+					"tra*.yaml",
+				},
+			},
+			expected: expected{
+				transformers: []string{
+					"do/not/deleteme/please.yaml",
+				},
+				deleted: []string{
+					"foo/foo.yaml",
+					"bar/bar.yaml",
+					"transformer3.yaml",
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			fSys := filesys.MakeFsInMemory()
+			testutils_test.WriteTestKustomizationWith(
+				fSys,
+				[]byte(fmt.Sprintf(
+					"transformers:\n  - %s", strings.Join(tc.given.transformers, "\n  - "))))
```

Comment:
ditto on the longer lines here and below, thanks

Reply:
Done and fixed in a few other spots.